libbeat/processors/add_cloud_metadata/provider_google_gce.go

```diff
@@ -121,7 +129,12 @@ var gceMetadataFetcher = provider{
 				}.ApplyTo(out, project)
 			}
 
-			return out
+			meta := common.MapStr{}
+			// call extractECSField for all ECS root fields like orchestrator.*
+			extractECSField("orchestrator", out, meta)
+
+			meta.DeepUpdate(common.MapStr{"cloud": out})
```

Comment:
It still seems that some special handling is done for `orchestrator` vs `cloud` fields. And it may be a bit confusint to have an `out`, that is not an output of the function at the end.

Wdyt about setting the values directly on `meta`, so they don't have to be moved from `out` to `meta`?

And in the same way, perhaps `cloud` fields could be directly set on `meta["cloud"]`, so this intermediary `out` is not needed.

Reply:
👍🏼  Good point! I tried to get rid of the extra "cooking" and applied the schema for orchestrator fields on a different map. Unfortunately `Apply` cannot work in `DeepUpdate` way so it's not possible to write directly to `cloud.*` in multiple places since `cloud*` will be overridden each time `Apply` is called.